lib/rbs/test/type_check.rb

```diff
@@ -151,6 +152,10 @@ def zip_args(args, fun, &block)
         end
       end
 
+      def values(vals, type)
```

Comment:
I'm not sure if extracting the _test with sampling_ to a method makes sense.
I think the `Hash` sampling would be like:

```rb
val.keys.sample(size).all? {|k| v = val[k]; value(...) }
```

Which cannot be used with the `values` method.

I'd like to suggest that trying writing sampling for `Hash` and then do refactoring.